File: Clase1_Hello_World_Java/app/src/main/java/com/luocz/clase1_hello_world_java/MainActivity2.java
package com.luocz.clase1_hello_world_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.RadioButton;
import android.widget.TextView;
import android.content.Intent;
import android.widget.Toast;

public class MainActivity2 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        Intent intent=getIntent();
        String message=intent.getStringExtra("Contra");
        TextView textView=findViewById(R.id.textView2);
        Toast.makeText(this, message,Toast.LENGTH_LONG).show();
        textView.setText(message);
    }

    public void actividad(View view) {
        Intent intent=new Intent(this,MainActivity3.class);
        startActivity(intent);
        finish();
    }

    public void clickMale(View view) {
        boolean checked=((RadioButton)view).isChecked();
        switch(view.getId()){
            case R.id.radioButton:
                if (checked)
                    Toast.makeText(this,"Hombre",Toast.LENGTH_LONG).show();
                    break;
            case R.id.radioButton2:
                if (checked)
                    Toast.makeText(this,"Mujer",Toast.LENGTH_LONG).show();
                    break;
        }
    }

    public void clickOnHobby(View view) {
        //primera forma
        boolean checked=((CheckBox)view).isChecked();
        //segunda forma
        CheckBox checkBox1=findViewById(R.id.checkBox1);
        CheckBox checkBox2=findViewById(R.id.checkBox2);

        boolean checked1=checkBox1.isChecked();
        boolean checked2=checkBox2.isChecked();
        switch(view.getId()){
            case R.id.checkBox1:
                //checke1
                if(checked){
                    Toast.makeText(this, "Se ha marcado en Deporte",Toast.LENGTH_LONG).show();
                }else{
                    Toast.makeText(this, "Se ha desmarcado  en Deporte",Toast.LENGTH_LONG).show();
                }

                break;
            case R.id.checkBox2:
                //checke2
                if(checked){
                    Toast.makeText(this,"Se ha marcado en No Deportes",Toast.LENGTH_LONG).show();
                }else{
                    Toast.makeText(this,"Se ha desmarcado en No Deportes",Toast.LENGTH_LONG).show();
                }
                break;
            default:
                break;
        }
    }
}

File: Clase1_Hello_World/app/src/main/java/com/Luocz/helloworld/MainActivity2.kt
package com.Luocz.helloworld

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.CheckBox
import android.widget.RadioButton
import android.widget.TextView
import android.widget.Toast

class MainActivity2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        val textview: TextView =findViewById(R.id.textView2)
        val message = intent.getStringExtra("Contra")
        Toast.makeText(this, message,Toast.LENGTH_LONG).show()
        Log.i("CICLO", message.toString())
        textview.setText(message.toString())
    }

    fun clickOnHobby(view: android.view.View) {
        val checkBox:CheckBox=view as CheckBox
        val contra="hola"
        checkBox.setText(contra)
        val checked:Boolean=checkBox.isChecked
        when(view.id){
            R.id.checkBox1->{
                if(checked){
                    Toast.makeText(this, "Se ha marcado en Deporte",Toast.LENGTH_LONG).show()
                }else{
                    Toast.makeText(this, "Se ha desmarcado  en Deporte",Toast.LENGTH_LONG).show()
                }

            }
            R.id.checkBox2->{
                if(checked){
                    Toast.makeText(this,"Se ha marcado en No Deportes",Toast.LENGTH_LONG).show()
                }else{
                    Toast.makeText(this,"Se ha desmarcado en No Deportes",Toast.LENGTH_LONG).show()
                }

            }
        }
    }

    fun clickMale(view: android.view.View) {
        val radioButtonseleccionado:RadioButton=view as RadioButton
        when(view.id){
            R.id.radioButton->{
                Toast.makeText(this,"Hombre",Toast.LENGTH_LONG).show()
            }
            R.id.radioButton2->{
                Toast.makeText(this,"Mujer",Toast.LENGTH_LONG).show()
            }
        }
    }

    fun actividad(view: android.view.View) {
        val intent = Intent(this, MainActivity3::class.java).apply {
        }
        startActivity(intent)
    }
}